File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Particulate Matter Bricklet

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletParticulateMatter pm =
		  new BrickletParticulateMatter(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current PM concentration
		int pm10, pm25, pm100;
		pm.GetPMConcentration(out pm10, out pm25, out pm100);

		Console.WriteLine("PM 1.0: " + pm10 + " µg/m³");
		Console.WriteLine("PM 2.5: " + pm25 + " µg/m³");
		Console.WriteLine("PM 10.0: " + pm100 + " µg/m³");

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, particulate_matter_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Particulate Matter Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let pm = ParticulateMatterBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current PM concentration.
    let pm_concentration = pm.get_pm_concentration().recv()?;

    println!("PM 1.0: {} µg/m³", pm_concentration.pm10);
    println!("PM 2.5: {} µg/m³", pm_concentration.pm25);
    println!("PM 10.0: {} µg/m³", pm_concentration.pm100);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
